azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -34,13 +35,14 @@ func resourceArmKubernetesCluster() *schema.Resource {
 				profile := rawProfiles[0].(map[string]interface{})
 				networkPlugin := profile["network_plugin"].(string)
 
-				if networkPlugin == "kubenet" {
+				if networkPlugin == "kubenet" || networkPlugin == "azure" {
 					dockerBridgeCidr := profile["docker_bridge_cidr"].(string)
 					dnsServiceIP := profile["dns_service_ip"].(string)
 					serviceCidr := profile["service_cidr"].(string)
 
-					if dockerBridgeCidr == "" || dnsServiceIP == "" || serviceCidr == "" {
-						return fmt.Errorf("If the `network_plugin` is set to `kubenet` then the fields `docker_bridge_cidr`, `dns_service_ip` and `service_cidr` must not be empty.")
+					if !((dockerBridgeCidr == "" && dnsServiceIP == "" && serviceCidr == "") ||
+						(dockerBridgeCidr != "" && dnsServiceIP != "" && serviceCidr != "")) {
+						return fmt.Errorf("`docker_bridge_cidr`, `dns_service_ip` and `service_cidr` should all be empty or neither should be empty.")
```

Comment:
This is a confusing error msg, could we change it to something like:
`docker_bridge_cidr`, `dns_service_ip` and `service_cidr` should all be empty or all should be set.

Reply:
Good suggestion, updated it.

---
In reply to: [207693716](https://github.com/terraform-providers/terraform-provider-azurerm/pull/1715#discussion_r207693716) [](ancestors = 207693716)